client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala

```diff
@@ -855,7 +881,8 @@ class LifecycleManager(val appUniqueId: String, val conf: CelebornConf) extends
                 s"unexpected! unknown appShuffleId $appShuffleId when checking shuffle deterministic level"))
         }
       } else {
-        shuffleIds.values.map(v => v._1).toSeq.reverse.find(isAllMaptaskEnd) match {
+        shuffleIds.values.filter(v => !isBarrierStage || v._2).map(v => v._1).toSeq.reverse.find(
```

Comment:
Why is there a check for `!isBarrierStage` here? It seems that there is no difference between the barrier stage and the indeterminate stage here. In other words, do we need to do a special check for the indeterminate stage?

Reply:
I agree, this should be fine - though I was minimizing the imapct of the change.